docs/src/architecture/data-model.md

```diff
@@ -0,0 +1,253 @@
+  | **Behavoiur**        | On cloud request<br/>-  provided configuration files are requested from the device and sent to the cloud<br/>- or downloaded from the cloud and sent to the device.<br/><br/> For details see [Configuration Managenement documentation](../references/c8y-configuration-management.md#configuration-files-for-child-devices)
+
+Examples **capability** objects for schema `tedge_config`:
+```json
+"tedge_config": {
+    "files": [
+        { "path": "/etc/tedge/tedge.toml", "type": "tedge.toml" },
+        { "path": "/etc/tedge/mosquitto-conf/c8y-bridge.conf" },
+        { "path": "/etc/tedge/mosquitto-conf/tedge-mosquitto.conf" },
+        { "path": "/etc/mosquitto/mosquitto.conf", "type": "mosquitto", "user": "mosquitto", "group": "mosquitto", "mode": "0o644" }
+    ]
+}
+```
+```json
+"tedge_config": {
+    "files": [
+        { "type": "foo.conf" },
+        { "type": "bar.conf" },
+    ]
+}
+```
+
+* Capability Schema: **Logging Management**
+
+  |                      |              | 
+  |:---------------------|:-------------|
+  | **Unique name**      | `tedge_log`  |
+  | **Field:** `files`   | TODO |
+  | **Behavoiur**        | TODO |
+
+
+## Inventory API
+
+The inventory is reflected on the MQTT bus under the topic `tedge/inventory`.
+
+* Each device object has it own topic: `/tedge/inventory/<device id>`
+* The `device id` is the `childid` of the **child-device** object, or `main` for the **thin-edge device** object.
+* The payload contains all fields of the device object in JSON format.
+* Example:
+   * topic: `/tedge/inventory/main`
+   * payload:
+```json
+     {
+        "name": "thin-edge device",
+        "type": "thin-edge.io"   
+     }
+```
+* The next level of the topic structure containes the **capability** objects per device:<br/>
+  `/tedge/inventory/<device id>/<capability type>`
+* Example:
+   * topic: `/tedge/inventory/main/tedge_config`
+   * payload:
+```json
+     {
+        "files": [
+           { "path": "/etc/tedge/tedge.toml", "type": "tedge.toml" },
+           { "path": "/etc/tedge/mosquitto-conf/c8y-bridge.conf" },
+           { "path": "/etc/tedge/mosquitto-conf/tedge-mosquitto.conf" },
+           { "path": "/etc/mosquitto/mosquitto.conf", "type": "mosquitto", "user": "mosquitto", "group": "mosquitto", "mode": "0o644" }
+        ]
+     }
+```
+* All messages to `tedge/inventory` and below are published as retain messages.
+  So one who is interested in any object of the inventory can just subscribe to the object's topic and gets directly the object, if it is available in the inventory.
+
+
+## Registration of a new device
+
+The sequence diagram below illustrates the data/message flow and all components involved, when a new external child-device registers it-self to thin-edge.
+
+```mermaid
+sequenceDiagram
+    participant external child device
+    participant tedge agent
+    participant inventory on MQTT
+    participant C8Y cfg plugin
+    participant C8Y log plugin
```

Comment:
That log plugin could be excluded to keep the diagram simpler as its behaviour is very similar to the config plugin.